app/models/user.rb

```diff
@@ -41,6 +41,20 @@ def salary
     salaries.last.try(:amount) || 0
   end
 
+  def salary_type
+    salaries.last.try(:type)
+  end
+
+  def count_day_offs_this_month
+    duration = 0
+    @vacations = vacations.all.approved.where("type" == "unpaid day offs").where("start_date >= ?", Date.today.beginning_of_month)
```

Comment:
1. it is better to use `self` for object methods, just not to mix them with private vars.
2. you don't need `all`, you can combine two `where`